addons/im_livechat/tests/test_im_livechat_channel.py

```diff
@@ -19,3 +20,16 @@ def test_operator_join_leave_livechat_channel(self):
         self.assertIn(bob_operator, self.livechat_channel.user_ids)
         self.livechat_channel.with_user(bob_operator).action_quit()
         self.assertNotIn(bob_operator, self.livechat_channel.user_ids)
+
+    def test_leave_livechat_channels_when_operator_access_revoked(self):
+        bob_operator = new_test_user(
+            self.env, "bob_user", groups="im_livechat.im_livechat_group_user"
+        )
+        self.assertNotIn(bob_operator, self.livechat_channel.user_ids)
+        self.livechat_channel.with_user(bob_operator).action_join()
+        self.assertIn(bob_operator, self.livechat_channel.user_ids)
+        livechat_operator_group = self.env.ref("im_livechat.im_livechat_group_user")
+        bob_operator.write({
+            "group_ids": [Command.unlink(livechat_operator_group.id)],
+        })
+        self.assertNotIn(bob_operator, self.livechat_channel.user_ids)
```

Comment:
should have a test for removing the manager group too
and the same kind of test but for removing users group group rather than group from users